### PR TITLE
AKS: Fix the roles to be included for Elastic SAN before installing Azure Container Storage

### PR DESCRIPTION
**Pull request by mukhoakash**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by mukhoakash** [comment]
[Live tests](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=90927511&view=results) queued:
- test_aks_create_with_azurecontainerstorage
- test_aks_update_with_azurecontainerstorage

**Review by FumingZhang — APPROVED** [review]

**Comment by FumingZhang** [comment]
> [Live tests](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=90927511&view=results) queued:
> 
> * test_aks_create_with_azurecontainerstorage
> * test_aks_update_with_azurecontainerstorage

One of the scenario tests `test_aks_update_with_azurecontainerstorage` failed for reasons unrelated to this change and will be fixed in #7512.

File: src/aks-preview/azext_aks_preview/azurecontainerstorage/_helpers.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def perform_role_operations_on_managed_rg(
     assign
 ):
     managed_rg_role_scope = build_role_scope(node_resource_group, None, subscription_id)
-    roles = ["Azure Container Storage Operator", "Reader"]
+    roles = ["Azure Container Storage Operator"]
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/azurecontainerstorage/_helpers.py:66]: May I ask will this change cause breaking changes?

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/azurecontainerstorage/_helpers.py:66]: This won't be a breaking change as the reader role is not necessary.

File: src/aks-preview/HISTORY.rst [hunk 2]
```diff
@@ -11,6 +11,9 @@ To release a new version, please select a new version number (usually plus 1 to
 
 Pending
 +++++++
+3.0.0b4
++++++++
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:15]:
```suggestion
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**